tds/src/main/java/thredds/server/wfs/WFSController.java

```diff
@@ -49,14 +55,137 @@ private void getCapabilities(PrintWriter out, HttpServletRequest hsreq) {
 	@RequestMapping("**")
 	public void httpHandler(HttpServletRequest hsreq, HttpServletResponse hsres) {

```

Comment:
I like that this method comprehensively addresses parameter parsing and validation including several edge cases.

In general, the longer the method, the more difficult it is to test. In general, the longer the method, the more likely that the method is doing reasoning at multiple levels of abstraction, which can make it harder for a subsequent programmer to reason about the method. Since this method is getting longer, it is worth asking: How could this method be broken up into smaller, more focused methods?

At a hand-waving level, I've seen web service layer classes successfully keep their methods smaller and more focused by separating the following concerns: request parameter parsing, request parameter validation, request dispatching/execution.

You might draw some more concrete inspiration from the analogous [method (doGet) in WMSServlet](https://github.com/Reading-eScience-Centre/edal-java/blob/0ec0dd7ed3480c56913b91d4a026eaa420636507/wms/src/main/java/uk/ac/rdg/resc/edal/wms/WmsServlet.java#L288-L343) in edal-java, though you needn't strictly follow the approach that they use.

Reply:
Yeah, We could have the checks as a separate function.